### DISCUSSION TITLE
Entry-point

### DISCUSSION THREAD

**Discussion by L4stR1t3s** [carbon-language/carbon-lang#2809] [Language design]
I just noticed that "Run" has been chosen as the entry-point? I understand the use of a verb is preferred but I don't expect a function named "Run" to act as an entry-point.

"Run" is generally used in the context of starting a program, process, ... but not the actual execution of it. I don't expect a function named "Run" to wait for the execution to finish before returning. I expect a function named "Do" to wait for the execution to finish, and then return. Not advocating to use "Do" as entry-point, just pointing out that different verbs create different expectations because of their prior use.

**Comment by jonmeow** [comment]
Just adding some context: https://github.com/carbon-language/carbon-lang/issues/1869 was the leads question, and https://github.com/carbon-language/carbon-lang/issues/1869#issuecomment-1449285904 has their rationale. I don't think that other possible meanings of `Run` were discussed there, I only see it for other possible names (`Start`, `Exec`, etc).
